Mimir/GraphQL/Types/AgentType.cs

```diff
@@ -47,4 +63,54 @@ protected override void Configure(IObjectTypeDescriptor<AgentObject> descriptor)
                     .ToList();
             });
     }
+
+    private static (
+        AgentRepository agentRepo,
+        PlanetName planetName,
+        Address agentAddress)? GetSource(IResolverContext context)
+    {
+        var agentRepo = context.Services.GetService<AgentRepository>();
+        if (agentRepo is null)
+        {
+            return null;
+        }
+
+        var planetName = context.ScopedContextData.TryGetValue("planetName", out var pn)
+            ? (PlanetName?)pn
+            : null;
+        if (planetName is null)
+        {
+            return null;
+        }
+
+        var agentAddress = context.Parent<AgentObject>().Address;
+        return (agentRepo, planetName.Value, agentAddress);
+    }
+
+    private static Agent? GetAgent(IResolverContext context)
```

Comment:
The returned value, `null` means too many meanings.
- There was no `AgentRepository` service.
- There was no given `planetName` argument.
- There was really no such agent.

It makes hard to know what happens. It should be more simple.

Reply:
You mean the side that sent the query should be able to tell why `null` is returned?

Reply:
I made issue #129 and move forward.